bin/util.js

```diff
@@ -0,0 +1,130 @@
+const _ = require('lodash');
+
+module.exports = {
+
+    /**
+     * Extract selected options in the provided command.
+     * Omits commander private variables and other objects.
+     *
+     * @param {Object} command - Commander.Command Instance
+     * @returns {Object} - Extracted options from command
+     */
+    commanderToObject: (command) => {
+        let obj = {
+            command: command.name(),
+            version: command.parent._version || false
+        };
+
+        // Iterate over all the command properties
+        _.forEach(Object.keys(command), (prop) => {
+            // Exclude command's private `_` variables and other objects
+            if (!_.startsWith(prop, '_') && !_.includes(['commands', 'options', 'parent'], prop)) {
+                obj[prop] = command[prop];
+            }
+        });
+
+        return obj;
+    },
+
+    /**
+     * Remove nested options having the provided prefix from `process.argv`.
+     *
+     * @param {Array} argv - Argument vector.
+     * @param {String} optionPrefix - Argument prefix to search for.
+     * @returns {Array} - All arguments without prefixed options and their values
+     */
+    omitNestedOptions: (argv, optionPrefix) => {
+        let args = [],
+            len,
+            i;
+
+        for (i = 0, len = argv.length; i < len; i++) {
+            if (!_.startsWith(argv[i], optionPrefix)) {
+                args.push(argv[i]);
+                continue;
+            }
+
+            // For prefixed args also omit its value, --prefix-arg value
+            if (argv[i + 1] && !_.startsWith(argv[i + 1], '-')) {
+                ++i;
+            }
+        }
+
+        return args;
+    },
+
+    /**
+     * Parse nested options having the provided prefix from `process.argv`.
+     *
+     * @param {Array} argv - Argument vector.
+     * @param {String} optionPrefix - Argument prefix to search for.
+     * @param {Array} options - Selected options.
+     * @returns {Object} Parsed object with nested options.
+     *
+     * @example
+     * let argv = ['--reporters=json,html', '--reporter-html-template=template.hb', '--reporter-export=path'],
+     *     options = ['json', 'html'];
+     * parseNestedOptions(argv, '--reporter-', options);
+     * //returns
+     * {
+     *   _generic: { export: path },
+     *   html: { template: template.hb },
+     *   json: {}
+     * }
+     *
+     */
+    parseNestedOptions: (argv, optionPrefix, options) => {
+        let args = [],
+            parsed = { _generic: {} },
+            name,
+            path,
+            len,
+            arg,
```

Comment:
`arg` is not being used outside the for loops below, it should be declared individually in both of them.